cookbooks/aws-parallelcluster-slurm/files/default/head_node_slurm/slurm/config_renderer.py

```diff
@@ -0,0 +1,177 @@
+
+    def render(self):
+        """Launch the rendering process."""
+        config = ""
+        if self.static_nodes > 0:
+            config += f"NodeName={self.static_node_name()}{ self._definitions() }\n"
+        if self.dynamic_nodes > 0:
+            config += f"NodeName={self.dynamic_node_name()}{ self._definitions(dynamic=True) }\n"
+
+        return config
+
+    def as_node_set(self):
+        """Alternative rendering for the NodeSet definition."""
+        node_set = ""
+        if self.static_nodes > 0:
+            node_set += f"{self.static_node_name()},"
+        if self.dynamic_nodes > 0:
+            node_set += f"{self.dynamic_node_name()},"
+
+        return node_set
+
+    def as_gres(self):
+        """Alternative rendering for the gres config."""
+        gres_render = ""
+        if self.gpu_count > 0:
+            if self.static_nodes > 0:
+                gres_render += (
+                    f"NodeName={self.static_node_name()} Name=gpu "
+                    f"Type={ self.gpu_type } File=/dev/nvidia[0-{ self.gpu_count - 1 }]\n"
+                )
+            if self.dynamic_nodes > 0:
+                gres_render += (
+                    f"NodeName={self.dynamic_node_name()} Name=gpu "
+                    f"Type={ self.gpu_type } File=/dev/nvidia[0-{ self.gpu_count - 1 }]\n"
+                )
+
+        return gres_render
+
+    def _definitions(self, dynamic=False):
+        definitions = f" CPUs={ self.vcpus() } RealMemory={ self.real_memory } State=CLOUD {self._features(dynamic)}"
+
+        if self.has_gpu and self.gpu_count > 0:
+            definitions += f" Gres=gpu:{ self.gpu_type }:{self.gpu_count}"
+
+        return definitions
+
+    def _features(self, dynamic=False):
+        resource_type = "static"
+        if dynamic:
+            resource_type = "dynamic"
+
+        instance_type = f",{self.instance_types[0]}"
+        if len(self.instance_types) > 1:
+            # When multiple instance types are defined we do not know in advance which one will be used
+            # to launch the node. So we do not list any of them as feature
+            instance_type = ""
+
+        features = f"Feature={resource_type}{instance_type},{self.name}"
+        if self.EFA_enabled:
+            features += ",efa"
+
+        if self.gpu_count > 0:
+            features += ",gpu"
+
+        return features
+
+    def _node_name(self, type, size):
+        return f"{self.queue_name}-{type}-{self.name}-[1-{ size }]"
+
+
+class QueueRenderer:
+    """Renders a PCluster Queue Config as a Slurm partition config or as gres config."""
+
+    def __init__(self, queue_config, no_gpu, memory_ratio, instance_types_data, conf_type="partition", default=False):
+        self.name = queue_config["Name"]
+        self.is_default = default
+        self.conf_type = conf_type
+        self.capacity_type = queue_config["CapacityType"]
+        self.iam = queue_config["Iam"]
```

Comment:
I assume we can remove this, right?

Reply:
I also think so.
I added them since they were read before and I expected to need them for some tests, but since the aren't yet used we can surely remove them.

Reply:
Done